estate/models/estate_property.py

```diff
@@ -0,0 +1,87 @@
+from dateutil.relativedelta import relativedelta
+from odoo import api, fields, models
+from odoo.exceptions import ValidationError
+from odoo.tools.float_utils import float_compare, float_is_zero
+
+class EstateProperty(models.Model):
+    _name = "estate.property"
+    _description = "Estate Property"
+
+    name = fields.Char()
+    description = fields.Text()
+    postcode = fields.Char()
+    date_availability = fields.Date(default=lambda self: fields.Date.today() + relativedelta(months=+3))
+    expected_price = fields.Float(required=True)
+    selling_price = fields.Float(required=True)
+    bedrooms = fields.Integer(default=2)
+    living_area = fields.Integer()
+    facades = fields.Integer()
+    garage = fields.Boolean()
+    garden = fields.Boolean()
+    garden_area = fields.Integer()
+    garden_orientation = fields.Selection(
+        string='Type',
+        selection=[('north', 'North'), ('south', 'South'), ('east', 'East'), ('west', 'West') ],
+        help="")
+    state = fields.Selection(
+        string='State',
+        selection=[('new', 'New'), ('offer_received', 'Offer Received'), ('offer_accepted', 'Offer Accepted'), ('sold', 'Sold'), ('canceled', 'Canceled') ],
+        default='new',
+        help="")
+    active = fields.Boolean(default=1)
+    property_type_id = fields.Many2one("estate.property.type")
+    seller_id = fields.Many2one("res.partner", string="Seller")
+    buyer_id = fields.Many2one("res.partner", string="Buyer")
+    tag_ids = fields.Many2many("estate.property.tag", string="Tags")
```

Comment:
In most cases it is not necessary, but not that on M2m fields we can define manually the name of the table used in between the 2 models. It is especially useful when multiple m2m fields exists between the same models. See relation in
https://www.odoo.com/documentation/17.0/developer/reference/backend/orm.html#odoo.fields.Many2many

In this particular case it's not an issue